src/com/luxoft/java/course/shevtsov/homework/algorithm/additionals/QuickSort.java

```diff
@@ -0,0 +1,49 @@
+package com.luxoft.java.course.shevtsov.homework.algorithm.additionals;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+
+public class QuickSort {
+
+    public static int[] sort(int[] ints) {
+        int[] ints1 = Arrays.copyOf(ints, ints.length);
+        recursSort(ints1, 0, ints1.length - 1);
+        return ints1;
+    }
+
+    private static void recursSort(int[] ints, int firstIndex, int lastIndex) {
```

Comment:
try to avoid use recursion

Reply:
Ok. Should I try to avoid recursion at all? Is recursion a not good practice? Is it because of memory spending?

Reply:
Yep

Reply:
done